packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue

```diff
@@ -54,93 +54,126 @@
   </div>
 </template>
 <script lang="ts">
-import { mapActions, mapGetters, mapMutations } from 'vuex'
 import { DavPermission } from '@ownclouders/web-client/src/webdav/constants'
-import { formatRelativeDateFromHTTP, formatFileSize } from '@ownclouders/web-pkg'
-import { defineComponent, inject, ref, Ref } from 'vue'
+import {
+  formatRelativeDateFromHTTP,
+  formatDateFromJSDate,
+  formatFileSize,
+  useClientService,
+  useDownloadFile,
+  useStore
+} from '@ownclouders/web-pkg'
+import { computed, defineComponent, inject, Ref, unref, watch } from 'vue'
 import { isShareSpaceResource, Resource, SpaceResource } from '@ownclouders/web-client/src/helpers'
 import { SharePermissions } from '@ownclouders/web-client/src/helpers/share'
-import { useClientService } from '@ownclouders/web-pkg/src/composables'
-import { formatDateFromJSDate } from '@ownclouders/web-pkg'
-import { useDownloadFile } from '@ownclouders/web-pkg'
+import { useTask } from 'vue-concurrency'
+import { useGettext } from 'vue3-gettext'
 
 export default defineComponent({
   name: 'FileVersions',
-  setup() {
-    const clientService = useClientService()
-    const loading = ref(false)
-
-    return {
-      ...useDownloadFile(),
-      space: inject<Ref<SpaceResource>>('space'),
-      resource: inject<Ref<Resource>>('resource'),
-      clientService,
-      loading
+  props: {
+    isReadOnly: {
+      type: Boolean,
+      required: false,
+      default: false
     }
   },
-  computed: {
-    ...mapGetters('Files', ['versions']),
-    hasVersion() {
-      return this.versions.length > 0
-    },
-    isRevertable() {
-      if ((this.space && isShareSpaceResource(this.space)) || this.resource.isReceivedShare()) {
-        if (this.resource.permissions !== undefined) {
-          return this.resource.permissions.includes(DavPermission.Updateable)
+  setup(props) {
+    const store = useStore()
+    const clientService = useClientService()
+    const { current: currentLanguage } = useGettext()
+    const { downloadFile } = useDownloadFile({ store, clientService })
+
+    const space = inject<Ref<SpaceResource>>('space')
+    const resource = inject<Ref<Resource>>('resource')
+
+    const versions = computed(() => {
+      return store.getters['Files/versions']
+    })
+
+    const fetchVersionsTask = useTask(function* () {
+      yield store.dispatch('Files/loadVersions', {
+        client: clientService.webdav,
+        fileId: unref(resource).fileId
+      })
+    })
+    const areVersionsLoading = computed(() => {
+      return !fetchVersionsTask.last || fetchVersionsTask.isRunning
+    })
+    watch(
```

Comment:
Unmounted wasn't possible ? :(

Reply:
Can either do a: `onMounted` AND a `watch` (without `immediate = true`) or b: only the `watch` with `immediate = true`. I usually decide for b in such a case. Happy about suggestions

Reply:
No that's fine, if we can't omit the watcher. Possible could also be a :key=resource.id on the respective panel and a onMounted, I prefer this over a watcher. But only if possible, otherwise looks nice to me

Reply:
Oh no, the watcher actually does more than watching the id. If you have the right sidebar in versions panel open from within an editor (e.g. text editor), the panel updates the versions list when you save the file. :-) but you actually gave me a different idea now, I switched over to only watching the `etag` of the resource, not the whole resource.